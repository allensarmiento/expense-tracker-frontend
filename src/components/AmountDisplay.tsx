import styled from 'styled-components';

type ColorProps = {
  color: string;
};

const Div = styled.div<ColorProps>`
  align-self: flex-start;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: .6rem 0;
  padding: .6rem .8rem;
  background-color: ${props => props.color === 'green'
    ? 'rgba(0, 128, 0, .5)' : 'rgba(255, 0, 0, .5)'};
  border-radius: .4rem;
  box-shadow: 0 .1rem .4rem rgba(0, 0, 0, .7);
`;

const Amount = styled.span<ColorProps>`
  padding: .4rem .8rem;
  background-color: ${props => props.color === 'green'
    ? 'rgba(0, 128, 0, .7)' : 'rgba(222, 3, 3, .7)'};
  border-radius: .4rem;
  box-shadow: 0 .1rem .4rem rgba(0, 0, 0, .7);
`;

type AmountDisplayProps = {
  amount: number | string;
  color: 'green' | 'red';
};

const AmountDisplay = ({ amount, color }: AmountDisplayProps) => {
  return (
    <Div color={color}>
      <span>{color === 'green' ? 'Income' : 'Expense'}:</span>&nbsp;
      <Amount color={color}>${amount}</Amount>
    </Div>
  );
};

export default AmountDisplay;
