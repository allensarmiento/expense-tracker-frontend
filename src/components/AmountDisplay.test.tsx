import { shallow } from 'enzyme';
import AmountDisplay from './AmountDisplay';

it('renders without crashing', () => {
  expect(shallow(<AmountDisplay amount={11.00} color="green" />))
    .toMatchSnapshot();

  expect(shallow(<AmountDisplay amount={20.00} color="red" />))
    .toMatchSnapshot();
});
