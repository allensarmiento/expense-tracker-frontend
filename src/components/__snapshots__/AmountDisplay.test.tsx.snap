// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div
  color="green"
>
  <span>
    Income
    :
  </span>
   
  <styled.span
    color="green"
  >
    $
    11
  </styled.span>
</styled.div>
`;

exports[`renders without crashing 2`] = `
<styled.div
  color="red"
>
  <span>
    Expense
    :
  </span>
   
  <styled.span
    color="red"
  >
    $
    20
  </styled.span>
</styled.div>
`;
